rfcs/20210119-determinism.md

```diff
@@ -0,0 +1,75 @@
+# RFC: Enabling Determinism in TensorFlow
+  
+| Status        | Proposed                                             |
+:---------------|:-----------------------------------------------------|
+| **Author(s)** | Pankaj Kanwar (Google), Reed Wanderman-Milne (Google)|
+| **Sponsor**   | Sanjoy Das (Google)                                |
+| **Updated**   | 2021-01-17                                           |
+
+
+## Objective
+Allow users to enable determinism behavior in TensorFlow. This means if the user runs a TensorFlow program multiple times, the model outputs and weights will be the same each time. Determinism will be supported on CPUs and GPUs.
+ 
+To get deterministic behavior, users must do the following:
+
+ * Enable determinism using the API proposed in this doc
+ * Use same hardware every run
+ * Use the same software environment every run (OS, checkpoints, version of TF, environmental variables, etc).
+ * Not use constructs outside TensorFlow that are non-deterministic, such as Python’s `random` module or using multiple threads/processes in ways that influence TensorFlow’s behavior
```

Comment:
A minor point, but worth getting right here and going forward: nondeterministic is spelled without a hyphen; see [Merriam-Webster](https://www.merriam-webster.com/dictionary/nondeterministic). It might be hyphenated in British English, but I'm not sure (and I'm British). Note that this word is used in several places in this document.